Sources/hostmgr-helper/VirtualMachineSlot.swift

```diff
@@ -19,121 +19,160 @@ class VirtualMachineSlot: NSObject, ObservableObject {
 
     enum Status: Sendable {
         case empty
-        case starting(LaunchConfiguration)
-        case running(LaunchConfiguration, IPv4Address)
-        case stopping
+        case starting(LaunchConfiguration, Task<ManagedVirtualMachine, Error>)
+        case running(ManagedVirtualMachine)
+        case stopping(LaunchConfiguration)
         case crashed(Error)
     }
 
+    enum Errors: Error {
+        /// the slot was asked to start a new VM when it wasn't available
+        case invalidStartState
+        /// the VM was stopped while starting
+        case vmStartCancelled
+    }
+
     struct ManagedVirtualMachine {
         let machine: VZVirtualMachine
         let config: LaunchConfiguration
+        let ip: IPv4Address
+
+        var handle: String {
+            config.handle
+        }
     }
 
     private let vmManager = VMManager()
 
     @Published
-    var managedVirtualMachine: ManagedVirtualMachine?
-
-    @Published @MainActor
     var status: Status = .empty
 
-    @Published @MainActor
-    var role: Role
+    let role: Role
 
     init(role: Role) {
         self.role = role
     }
 
-    @MainActor
     func start(launchConfiguration: LaunchConfiguration) async throws {
+        Logger.helper.log("Launch request for \(launchConfiguration.handle).")
 
-        self.status = .starting(launchConfiguration)
+        guard isAvailable else {
+            throw Errors.invalidStartState
+        }
 
         do {
-            let virtualMachine = try await launchConfiguration.setupVirtualMachine()
-            virtualMachine.delegate = self
-            self.managedVirtualMachine = ManagedVirtualMachine(machine: virtualMachine, config: launchConfiguration)
-
-            try await virtualMachine.start()
-
-            if launchConfiguration.waitForNetworking {
-                let ipAddress = try await vmManager.ipAddress(forVmWithName: launchConfiguration.handle)
-                Logger.helper.log("Startup complete – IP Address: \(ipAddress.debugDescription)")
-                self.status = .running(launchConfiguration, ipAddress)
-            } else {
-                Logger.helper.log("Startup in progress – skipped waiting for IP address per launch configuration")
-                self.status = .running(launchConfiguration, .any)
+            let startTask = Task { try await createManagedVm(launchConfiguration: launchConfiguration) }
```

Comment:
This `Task` should propagate down its cancellation, for instance when `Task.sleep` [is called](https://github.com/Automattic/hostmgr/blob/476f9020b234018bba3f701f3445d09c61a2fcb4/Sources/libhostmgr/Platforms/VMManager.swift#L205) resulting in a short closure lifespan when it's not needed.